### ISSUE TITLE
ruff / isort doesn't detect all duplicate imports

### ISSUE THREAD

**Issue by CharlesPerrotMinotHCHB** [astral-sh/ruff#10087]
Hi,

We do use `combine-as-imports` for the lint.isort settings. However, the following passes ruff check despite the library marshaling being imported twice:

```
import typing
from typing import Any

def test_function(my_input: Any) -> typing.Any:
  return my_input
```

My expectation would be for ruff to catch this as a duplicate import, and force the import simplification.

**Comment by MichaReiser** [comment]
> My expectation would be for ruff to catch this as a duplicate import, and force the import simplification.

I can see how unifying the imports is desired but it imposes a few challenges: 

The first is that it depends on personal preference on how the final code should look like. Should `typing.Any` be replaced with `Any` or the other way round where `Any` is replaced with `typing.Any`? What if there are multiple named imports and references to `typing`? What if there are no named imports for some usages of `typing`? I suspect that this heavily depends on personal preference and some users might not want that `typing` gets removed altogether. 

That's why I think this is something that's out of scope of the `isort` rule and remains something that requires manual intervention. There's also the technical challenge that it requires rewriting all usages in the file

**Comment by yoann9344** [comment]
There are also situations with as import.
```python
Import typing as t
Import typing as plop
...
plop.Any, t.Any
```

**Comment by CharlesPerrotMinotHCHB** [comment]
> There are also situations with as import.
> 
> ```python
> Import typing as t
> Import typing as plop
> ...
> plop.Any, t.Any
> ```
> 
> Maybe creating temporary errors, like TEMP, to throw the error without breaking the isort autofix. Until a more complexe décision to be taken.

Hum... to me this should not be authorized. Dual import doesn't make much sense in my opinion

**Comment by CharlesPerrotMinotHCHB** [comment]
> > My expectation would be for ruff to catch this as a duplicate import, and force the import simplification.
> 
> I can see how unifying the imports is desired but it imposes a few challenges:
> 
> The first is that it depends on personal preference on how the final code should look like. Should `typing.Any` be replaced with `Any` or the other way round where `Any` is replaced with `typing.Any`? What if there are multiple named imports and references to `typing`? What if there are no named imports for some usages of `typing`? I suspect that this heavily depends on personal preference and some users might not want that `typing` gets removed altogether.
> 
> That's why I think this is something that's out of scope of the `isort` rule and remains something that requires manual intervention. There's also the technical challenge that it requires rewriting all usages in the file

Fixing it would clearly be quite hard. But at least raise an error?
